packages/express/src/authenticateRequest.ts

```diff
@@ -95,3 +96,38 @@ const absoluteProxyUrl = (relativeOrAbsoluteUrl: string, baseUrl: string): strin
   }
   return new URL(relativeOrAbsoluteUrl, baseUrl).toString();
 };
+
+export const authenticateAndDecorateRequest = (options: ClerkMiddlewareOptions = {}) => {
+  const clerkClient = options.clerkClient || defaultClerkClient;
+  const enableHandshake = options.enableHandshake || false;
+
+  // eslint-disable-next-line @typescript-eslint/no-misused-promises
+  const middleware: RequestHandler = async (request, response, next) => {
```

Comment:
Have we covered for the scenario where someone uses both middlewares? A little conditional at the top of this helper to just call next if `auth` is already on the request I think would take care of that. May be worth adding a test for this as well.

Reply:
Nope, but added just now and a test too!

Reply:
Do we need to error if there are both? Could we just add a conditional check for `req.auth` at the top of the function instead?

Reply:
Ah I get you now, we want to allow usage of both, but check for `req.auth` first then call `next()` if exists so we dont have to run the same logic etc 👍🏼 Perfect for scenario like

```ts
app.use(clerkMiddleware());

app.get('/protected', requireAuth(), (req, res) => {
  res.send('This is a protected route');
});
```